sdk/core/core-http/src/credentials/accessTokenCache.ts

```diff
@@ -33,26 +33,39 @@ export interface AccessTokenCache {
  * passed.
  */
 export class ExpiringAccessTokenCache implements AccessTokenCache {
-  private tokenRefreshBufferMs: number;
+  private beforeTokenExpiresInMs: number;
   private cachedToken?: AccessToken = undefined;
 
   /**
    * Constructs an instance of {@link ExpiringAccessTokenCache} with
    * an optional expiration buffer time.
    */
   constructor(tokenRefreshBufferMs: number = TokenRefreshBufferMs) {
-    this.tokenRefreshBufferMs = tokenRefreshBufferMs;
+    this.beforeTokenExpiresInMs = tokenRefreshBufferMs;
   }
 
   setCachedToken(accessToken: AccessToken | undefined): void {
     this.cachedToken = accessToken;
   }
 
-  getCachedToken(): AccessToken | undefined {
-    if (
+  /**
+   * Returns true if the token hasn't yet expired, but it's within the expiration time period.
+   */
+  isReadyToRefresh(): boolean {
```

Comment:
but returns false if the token doesn't exist? So it's not ready to refresh?

It's really confusing to me what an external caller wants from this thing. It feels like we've overcomplicated the `AccessTokenCache` interface with a lot of extra information that is harder to reason about.

Reply:
Resolving since I reverted this.